src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -532,187 +897,241 @@ contract StrategyManagerUnitTests is Test, Utils {
-        strategyManager.addStrategiesToDepositWhitelist(_strategy);
-        cheats.stopPrank();
-
+    /**
+     * @notice testing removeShares() function with 3 strategies deposited.
+     * Removing Shares could result in removing from staker strategy list if depositAmounts[i] == sharesAmounts[i].
+     * Only callable by DelegationManager
+     */
+    function testFuzz_RemoveShares(uint256[3] memory depositAmounts, uint256[3] memory sharesAmounts) external {
         address staker = address(this);
-        IERC20 token = dummyToken;
-        uint256 amount = 1e18;
-        IStrategy strategy = dummyStrat;
+        IStrategy[] memory strategies = new IStrategy[](3);
+        strategies[0] = dummyStrat;
+        strategies[1] = dummyStrat2;
+        strategies[2] = dummyStrat3;
+        uint256[] memory sharesBefore = new uint256[](3);
+        for (uint256 i = 0; i < 3; ++i) {
+            cheats.assume(sharesAmounts[i] > 0 && sharesAmounts[i] <= depositAmounts[i]);
+            _depositIntoStrategySuccessfully(strategies[i], staker, depositAmounts[i]);
+            sharesBefore[i] = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            require(sharesBefore[i] == depositAmounts[i], "Staker has not deposited amount into strategy");
+            require(_isDepositedStrategy(staker, strategies[i]), "strategy should be deposited");
+        }
+        uint256 stakerStrategyListLengthBefore = strategyManager.stakerStrategyListLength(staker);
 
-        cheats.startPrank(staker);
-        cheats.expectRevert();
-        strategyManager.depositIntoStrategy(strategy, token, amount);
-        cheats.stopPrank();
+        uint256 numPoppedStrategies = 0;
+        uint256[] memory sharesAfter = new uint256[](3);
+        for (uint256 i = 0; i < 3; ++i) {
+            delegationManagerMock.removeShares(strategyManager, staker, strategies[i], sharesAmounts[i]);
+            sharesAfter[i] = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            if (sharesAmounts[i] == depositAmounts[i]) {
+                ++numPoppedStrategies;
+                require(
+                    !_isDepositedStrategy(staker, strategies[i]),
+                    "strategy should not be part of staker strategy list"
+                );
+                require(sharesAfter[i] == 0, "sharesAfter != 0");
+            } else {
+                require(_isDepositedStrategy(staker, strategies[i]), "strategy should be part of staker strategy list");
+                require(
+                    sharesAfter[i] == sharesBefore[i] - sharesAmounts[i],
+                    "sharesAfter != sharesBefore - sharesAmounts"
+                );
+            }
+        }
+        require(
+            stakerStrategyListLengthBefore - numPoppedStrategies == strategyManager.stakerStrategyListLength(staker),
+            "stakerStrategyListLengthBefore - numPoppedStrategies != strategyManager.stakerStrategyListLength(staker)"
+        );
```

Comment:
if we're bothering to check `_isDepositedStrategy` before, we should probably check this afterwords as well? i.e. verify that the correct strategy(ies) are removed, rather than just that the correct _number of strategies_ is removed

Reply:
Within the for loop after each `removeShares()` call its checking that `_isDepositedStrategy` is false when the strategy is removed. But I moved this into a separate loop after calling `removeShares()` in all strategies https://github.com/Layr-Labs/eigenlayer-contracts/pull/293/commits/f011f90397319e8de8784535dc15b08fe943d165